### DISCUSSION TITLE
Failing to publish to github - "Bad credentials []"

### DISCUSSION THREAD

**Discussion by shmul** [goreleaser/goreleaser#5001] [Q&A] [answered]
Hi,

I'm trying to run the first publish to github by running `goreleaser release --skip=validate --clean` but I always get this error regardless of whether `GITHUB_TOKEN` is defined or `env_files.github_token` is defined in the `.goreleaser.yaml` file.  Any help would be appreciated.
```
scm releases: failed to publish artifacts: could not release: POST https://api.github.com/repos/XXX/releases: 401 Bad credentials []
```

BTW, I'm using `--skip=validate` as I have untracked files in my repo that I don't want to add to `.gitignore` (yep, I realize it isn't a great practice). It would be nice to be able to ignore just untracked files rather than skipping the entire validation.

Thanks,
Shmul

**Comment by caarlos0** [comment, marked as answer]
that error is coming from github, which means your creds are invalid.


goreleaser should log which creds its using (e.g. it will log if its using `$GITHUB_TOKEN` or the env file), so, make sure its using the right one there.

**Reply by shmul** [reply]
Thanks. Indeed this was the cause, but I'll document it here in case others will encounter it as well.
I have a github token file whose contents were in the form `user:token` which is the format used in my docker build file in order to pull private modules
`RUN git config --global url."https://$(cat /etc/secrets/gh_token)@github.com/".insteadOf "https://github.com/"`

This format doesn't work for the API calls that goreleaser uses, so I generated a separate token (a good practice in any case).
